cmd/tikv-ctl/src/main.rs

```diff
@@ -1983,33 +1983,18 @@ fn main() {
         let key = unescape(&key);
         split_region(&pd_client, mgr, region_id, key);
     } else {
-        // Deal with all subcommands about db or host.
-        let mut contains_db = true;
-        let data_dir = opt.data_dir.unwrap_or_else(|| {
-            contains_db = false;
-            String::new()
-        });
-        let host = opt.host.unwrap_or_else(|| {
-            if !contains_db {
-                clap::Error {
-                    message: String::from("[host|data-dir] is not specified"),
-                    kind: ErrorKind::MissingRequiredArgument,
-                    info: None,
-                }
-                .exit();
+        let data_dir = opt.data_dir.as_deref();
+        let host = opt.host.as_deref();
```

Comment:
```suggestion
        let data_dir = opt.data_dir.filter(|s| !s.is_empty());
        let host = opt.host.filter(|s| !s.is_empty());
```

Reply:
But the following method's parameters are Option<&str>, I think it is better to use `deref` here rather than modify the following methods all.

Reply:
ok